includes/embeds/class-amp-core-block-handler.php

```diff
@@ -269,6 +280,161 @@ public function dequeue_block_library_file_script() {
 		wp_dequeue_script( 'wp-block-library-file' );
 	}
 
+	/**
+	 * Ampify navigation block contained by <nav> element.
+	 *
+	 * Steps:
+	 * - "fake" the "open" state by adding `is-menu-open has-modal-open` classes to `div.wp-block-navigation__responsive-container`,
+	 * - add `on="tap:{id}.open"` to `button.wp-block-navigation__responsive-container-open` element,
+	 * - add `on="tap:{id}.close"` to `button.wp-block-navigation__responsive-container-close` element,
+	 * - wrap `div.wp-block-navigation__responsive-container` with `<amp-lightbox id="{id}" layout="nodisplay">...</amp-lightbox>`,
+	 * - remove `data-micromodal-trigger` and `data-micromodal-close` attributes,
+	 * - remove `aria-expanded` and `aria-modal` attributes,
+	 * - duplicate `div.wp-block-navigation__responsive-container` (original one, without extra classes) outside the `amp-lightbox` wrapper and unwrap it from modal-related wrappers,
+	 * - dequeue the `wp-block-navigation-view` script.
+	 *
+	 * @see https://github.com/ampproject/amp-wp/issues/6319#issuecomment-978246093
+	 * @see render_block_core_navigation()
+	 *
+	 * @since 2.2
+	 *
+	 * @param string $block_content The block content about to be appended.
+	 * @param array  $block         The full block, including name and attributes.
+	 *
+	 * @return string Filtered block content.
+	 */
+	public function ampify_navigation_block( $block_content, $block ) {
+		add_action( 'wp_print_scripts', [ get_class(), 'dequeue_block_navigation_view_script' ], 0 );
+		add_action( 'wp_print_footer_scripts', [ get_class(), 'dequeue_block_navigation_view_script' ], 0 );
+
+		$dom = new Document();
```

Comment:
According to @westonruter comment on [#6319](https://github.com/ampproject/amp-wp/issues/6319#issuecomment-973273079):

> Note: Instead of a `render_block` filter to the `AMP_Core_Block_Handler` class (which will entail manipulating an HTML string), it may make sense instead to utilize a sanitizer instead where the DOM can be interacted with instead. If the `render_block` filter is clean, then we can go ahead with that.

I don't think we should be interacting with DOM directly in the `AMP_Core_Block_Handler`.

I'm also thinking if there's a simpler way to ampify the new navigation block. I don't think we actually need to use `amp-lightbox` or `amp-sidebar`. Why not rely on bare CSS classes and just use `amp-bind` and `amp-state` instead?

Reply:
Here's an alternative approach that's using `amp-bind` only as mentioned above:

https://github.com/ampproject/amp-wp/pull/6838

There's 1 todo item left: add support for opening submenus on click.